tests/storages_tests/rdb_tests/test_storage.py

```diff
@@ -267,6 +278,140 @@ def test_upgrade_multi_objective_optimization(optuna_version: str) -> None:
+        for trial in study.trials:
+            assert trial.system_attrs["a"] == 0
+            assert trial.user_attrs["b"] == 1
+            assert trial.intermediate_values[0] == 0.5
+            assert -5 <= trial.params["x1"] <= 5
+            assert 1e-5 <= trial.params["x2"] <= 1e-3
+            assert -6 <= trial.params["x3"] <= 6
+            assert 0 <= trial.params["y1"] <= 10
+            assert 1 <= trial.params["y2"] <= 20
+            assert 5 <= trial.params["y3"] <= 15
+            assert trial.params["z"] in (-5, 0, 5)
+            assert trial.value is not None and 0 <= trial.value <= 820
+
+        assert study.system_attrs["c"] == 2
+        assert study.user_attrs["d"] == 3
+
+
+@pytest.mark.parametrize("optuna_version", ["2.6.0"])
+def test_upgrade_multi_objective_optimization(optuna_version: str) -> None:
+    src_db_file = os.path.join(
+        os.path.dirname(__file__), "test_upgrade_assets", f"{optuna_version}.db"
+    )
+    with tempfile.TemporaryDirectory() as workdir:
+        shutil.copyfile(src_db_file, f"{workdir}/sqlite.db")
+        storage_url = f"sqlite:///{workdir}/sqlite.db"
+
+        storage = RDBStorage(storage_url, skip_compatibility_check=True)
+        assert storage.get_current_version() == f"v{optuna_version}.a"
+        head_version = storage.get_head_version()
+        storage.upgrade()
+        assert head_version == storage.get_current_version()
+
+        # Create a new study.
+        study = create_study(storage=storage, directions=["minimize", "minimize"])
+        assert len(study.trials) == 0
+        study.optimize(mo_objective_test_upgrade, n_trials=1)
+        assert len(study.trials) == 1
+
+        # Check empty study.
+        study = load_study(storage=storage, study_name="multi_empty")
+        assert len(study.trials) == 0
+        study.optimize(mo_objective_test_upgrade, n_trials=1)
+        assert len(study.trials) == 1
+
+        # Resume single objective optimization.
+        study = load_study(storage=storage, study_name="multi")
+        assert len(study.trials) == 1
+        study.optimize(mo_objective_test_upgrade, n_trials=1)
+        assert len(study.trials) == 2
+        for trial in study.trials:
+            assert trial.system_attrs["a"] == 0
+            assert trial.user_attrs["b"] == 1
+            assert -5 <= trial.params["x1"] <= 5
+            assert 1e-5 <= trial.params["x2"] <= 1e-3
+            assert -6 <= trial.params["x3"] <= 6
+            assert 0 <= trial.params["y1"] <= 10
+            assert 1 <= trial.params["y2"] <= 20
+            assert 5 <= trial.params["y3"] <= 15
+            assert trial.params["z"] in (-5, 0, 5)
+            assert -25 <= trial.values[0] <= 25
+            assert 0 <= trial.values[1] <= 820
+
+        assert study.system_attrs["c"] == 2
+        assert study.user_attrs["d"] == 3
+
+
+@pytest.mark.parametrize("optuna_version", ["2.6.0"])
+def test_upgrade_distributions(optuna_version: str) -> None:
```

Comment:
Isn't this distribution upgrade test something you want to do only with 2.6.0? I don't think parametrize is necessary.

Reply:
You're right. `objective_test_upgrade` contains a distribution that is not supported by old versions of Optuna to made me parametrize here. As you pointed out we should tests migration from other versions of Optuna so let me take time to improve.